examples/experimental/http-route-timeouts/timeout-example.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: timeout-example
+spec:
+  parentRefs:
+  - name: example-gateway
+  hostnames:
+  - "timeout.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /timeout
+    backendRefs:
+    - name: timeout-svc
+      port: 8080
+    timeouts:
+      request: 10s
+      backendRequest: 2s
```

Comment:
Non-blocking nits:

1. Would recommend omitting hostnames as they're not required for the example
2. Generally prefer to structure Route rules in the order of matches -> filters/timeouts/modifiers -> backends. In practice, that would mean moving timeouts above backendRefs in this example.